### DISCUSSION TITLE
FM Point Sources: Beta pic ground truth

### DISCUSSION THREAD

**Discussion by emrickman** [vortex-exoplanet/VIP#555] [Q&A] [answered]
Hello, I am trying to use the negative fake planet injection on a number of detected companions with SPHERE IRDIS, and I first want to understand the given example of beta pic before I proceed with my own data. 

The example as given in Jupyter Notebook "05_fm_planets" quotes the ground truth values of beta pic as:
r = 830.00+-4.56 mas (GT: 829.29 mas), 
PA = 239.95+-0.24 deg (GT: 240.00 deg) 
f = 394.13+-21.67 ADUs (GT: 400.00 ADUs)

But from reading Absil et al. (2013), my understanding is the values are:
Separation (mas): 452 ± 10
PA (deg): 211.2 ± 1.3
∆L = 8.01±0.16 mag

I have several questions related to this:

1. Why is there a difference between the 'ground truth' of 830mas and the published value of 452mas? And the difference between PA=239.95 and PA=211.2?
2.  Does the calculated flux i.e. 394.13, directly correspond to the contrast ratio between the primary star PSF and the companion flux? And related to this, does this take into account the difference in the transmission between the neutral density filters used for the flux frames and the coronagraphic frames?
3.  It's not clear to me whether the inserted cube needs to be before the rotation or the derotated cube? And is the first guess of the negative fake planet for the predicted final position? (i.e. the derotated cube position?)
4. What does the value for derot_off correspond to? For SPHERE IRDIS is that "HIERARCH ESO ADA ABSROT END" or "HIERARCH ESO ADA ABSROT START" or something different?

Many thanks for taking the time to clarify these questions, and I hope to be able to use this for my SPHERE IRDIS data.

**Comment by oabsil** [comment, marked as answer]
Hi @emrickman! @VChristiaens being currently at a conference, I'm trying to answer some of your questions here. I will let Valentin further comment if needed.
1. the ground truth does not correspond to the coordinates of beta Pic b, but to the fake companion that is injected earlier in the same notebook.
2. the calculated flux is a scaling factor for the normalized off-axis PSF. Converting it into a contrast requires access to the flux in the non-normalized off-axis PSF, to the ratio of the DITs used for science and PSF observations, and to the characteristics of any ND used for PSF observations.
3. as far as I know, all VIP routines use the original (calibrated) cube, before any derotation. This is why the parallactic angles are also always passed as argument.
4. derot_off was specifically computed for the NACO data used here, to correct the parallactic angles for instrumental effects. I'm not sure about the equivalent for SPHERE. Maybe Valentin can answer that.
Cheers,
Olivier

**Comment by VChristiaens** [comment]
Hi @emrickman, thanks for your interest in using VIP! And thanks @oabsil for your prompt reply! Just to provide some more details:
1. It is indeed not planet b, but a fake companion that is injected in the first part of tutorial 5, such that we can use the injected parameters as ground truth. Planet b is not seen in the images as we only applied PCA on a single annulus at larger separation.
2. As @oabsil said, the effects of instrument-/observation-specific parameters should be calculated separately -- in particular, difference in integration time between off-axis and coronagraphic sequence, or the effect of the ND filter on the off-axis PSF. Only the radial coronagraphic transmission profile can be passed as an optional input to the NEGFC routines. The contrast is found from the ratio of the flux returned by the NEGFC algorithm (modulo the above effects) with the stellar PSF flux. The latter can be inferred with the ``fm.normalize_psf`` routine applied to an offset PSF image.
3. The input should be the calibrated cube, and first estimate on the position should be that in your derotated image (e.g. median-ADI or PCA-ADI image)
4. For the SPHERE parallactic angles, you need to consider the values for "HIERARCH ESO TEL PARANG START" and "HIERARCH ESO TEL PARANG END" header keywords. You can interpolate between these 2 values for a more accurate value for each frame of your individual cubes. To get the final derotation angles needed to align your images with north up, you then need to add to the parallactic angles the true north correction, pupil offset and (if you use IFS data) the additional offset corresponding to the IFS instrument: -1.75, 135.99 and -100.48 deg, resp. (Maire et al. 2016).
Hopefully this helps.
Cheers,
Valentin
